spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -102,10 +106,13 @@
   private final StructType dsSchema;
   private final Map<String, String> extraSnapshotMetadata;
   private final boolean partitionedFanoutEnabled;
+  private final Distribution requiredDistribution;
+  private final SortOrder[] requiredOrdering;
 
   SparkWrite(SparkSession spark, Table table, SparkWriteConf writeConf,
              LogicalWriteInfo writeInfo, String applicationId,
-             Schema writeSchema, StructType dsSchema) {
+             Schema writeSchema, StructType dsSchema,
+             Distribution requiredDistribution, SortOrder[] requiredOrdering) {
```

Comment:
+1

In general, I think I agree with the choice to implement decisions in the write builder and simply pass in the results here. So building the distribution and ordering is up to the write builder, and this just returns the required distribution and ordering to Spark.

Reply:
Yeah, this class is already fairly complicated. Adding more logic would make it even worse.